File: desafio-02/caloni/cpp/desafio2.cpp
#include <iostream>
#include <list>

using namespace std;


int main()
{
    list<int> primes = { 2, 3, 5, 7 };

    for( int i = 11; i <= 10000; ++i ) 
    {
        bool is_prime = true;

        for( int prime: primes )
        {
            if( i % prime == 0 )
            {
                is_prime = false;
                break;
            }
        }

        if( is_prime )
            primes.push_back(i);
    }

    for( int prime: primes )
        cout << prime << endl;
}

File: desafio-02/luizfelmach/python/desafio2.py
# Python 3.9.9
"""System module."""
from math import ceil

def isPrime(number, div, count, primes):
    """Verify if number is prime."""
    if div <= ceil(number**0.5):
        isDiv = [prime for prime in primes if number % prime == 0]
        if (len(isDiv) >= 1 or number == 1):
            count += 1
        div += 1
        return isPrime(number, div, count, primes)
    return primes.append(number)

primesList = []

for i in range(1, 10000):
    isPrime(i, 1, 0, primes=primesList)
print(primesList)
